flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -795,108 +795,111 @@ extends Actor with ActorLogMessages with ActorLogging {
     var startRegisteringTask = 0L
     var task: Task = null
 
-    // all operations are in a try / catch block to make sure we send a result upon any failure
-    try {
-      // check that we are already registered
-      if (!isConnected) {
-        throw new IllegalStateException("TaskManager is not associated with a JobManager")
-      }
-      if (slot < 0 || slot >= numberOfSlots) {
-        throw new Exception(s"Target slot ${slot} does not exist on TaskManager.")
-      }
+    if (!isConnected) {
+      sender ! Failure(
+        new IllegalStateException("TaskManager is not associated with a JobManager.")
+      )
+    } else if (slot < 0 || slot >= numberOfSlots) {
+      sender ! Failure(new Exception(s"Target slot $slot does not exist on TaskManager."))
+    } else {
+      sender ! Acknowledge
 
-      val userCodeClassLoader = libraryCacheManager match {
-        case Some(manager) =>
-          if (LOG.isDebugEnabled) {
-            startRegisteringTask = System.currentTimeMillis()
-          }
+      Future {
+        try {
```

Comment:
Can we pull the code in the future into its own method? Makes it easier to understand by separating the different parts along the methods.